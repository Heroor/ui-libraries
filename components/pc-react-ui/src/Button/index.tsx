import React from 'react';
import { Button } from './Button';

const withIDE = (Component: React.ElementType) => {
  return React.forwardRef((props, ref) => {
    // TODO IDE 适配逻辑写在这...
    return (
      <Component
        {...props}
        ref={ref}
      />
    );
  });
};

const ButtonHoc = withIDE(Button);

export default ButtonHoc;
