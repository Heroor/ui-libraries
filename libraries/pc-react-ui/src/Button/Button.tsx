// eslint-disable-next-line import/no-extraneous-dependencies
import React, { FC, HTMLAttributes } from 'react';
import './button.css';

export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Button contents
   */
  label?: string;
}

/**
 * Primary UI component for user interaction
 */
export const Button: FC<ButtonProps> = ({
  primary, backgroundColor, size, label, ...rest
}) => {
  const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';
  return (
    <button
      {...rest}
      type="button"
      className={['storybook-button', `storybook-button--${size}`, mode].join(' ')}
      style={backgroundColor ? { backgroundColor } : {}}
    >
      {label}
    </button>
  );
};

Button.defaultProps = {
  backgroundColor: '',
  label: '',
  primary: false,
  size: 'medium',
};
