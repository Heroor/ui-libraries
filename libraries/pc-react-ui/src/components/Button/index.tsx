import { Button as AntdButton } from 'antd';
import type { ButtonProps as AntdButtonProps } from 'antd/lib/button';
import * as basicsPlugin from './plugins/basicsPlugin';
import type { pluginType } from '@/plugins/type';
import { Plugin, registerComponet } from '../../plugins/index';

// import './button.css';

type ButtonProps = AntdButtonProps

const mapProps = {
  mySize: 'size',
};
const plugin = new Plugin(basicsPlugin, { displayName: 'Button', mapProps });

const Button = registerComponet<ButtonProps, pluginType<ButtonProps>>(
  AntdButton,
  plugin,
);

export default Button;
