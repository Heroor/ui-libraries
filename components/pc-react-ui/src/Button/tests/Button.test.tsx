import { expect, it, describe } from 'vitest';
import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Test Button Render', () => {
  it('Normal Button', () => {
    const component = renderer.create(
      <Button>Button</Button>,
    );

    const result = component.toJSON();
    expect(result).toMatchSnapshot();
  });

  it('Primary Button', () => {
    const component = renderer.create(
      <Button primary>Button</Button>,
    );

    const result = component.toJSON();
    expect(result).toMatchSnapshot();
  });

  it('Size Button', () => {
    const component = renderer.create(
      <Button primary size="large">Button</Button>,
    );

    const result = component.toJSON();
    expect(result).toMatchSnapshot();
  });
});
