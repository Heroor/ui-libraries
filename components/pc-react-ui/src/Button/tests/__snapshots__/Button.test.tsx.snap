// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Button Render > Normal Button 1`] = `
<button
  className="storybook-button storybook-button--medium storybook-button--secondary"
  style={{}}
  type="button"
/>
`;

exports[`Test Button Render > Primary Button 1`] = `
<button
  className="storybook-button storybook-button--medium storybook-button--primary"
  style={{}}
  type="button"
/>
`;

exports[`Test Button Render > Size Button 1`] = `
<button
  className="storybook-button storybook-button--large storybook-button--primary"
  style={{}}
  type="button"
/>
`;
